lib/chef/provider/package/macports.rb

```diff
@@ -3,7 +3,7 @@ class Provider
     class Package
       class Macports < Chef::Provider::Package
 
-        provides :macports_package, os: "mac_os_x"
+        provides :macports_package, os: "darwin"
```

Comment:
this should probably have no filters... if someone actually specified `macports_package`, then they are overriding the normal logic. That seems to be how the other OS-specific providers with their own resource work... (like homebrew_package)